src/text_shaper/directwrite_analysis_wrapper.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * This file is part of the "libterminal" project
+ *   Copyright (c) 2019-2020 Christian Parpart <christian@parpart.family>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include<string>
+
+#include <dwrite.h>
+#include <dwrite_3.h>
+#include <wrl/implements.h>
+
+using Microsoft::WRL::RuntimeClass;
+using Microsoft::WRL::RuntimeClassFlags;
+using Microsoft::WRL::ClassicCom;
+using Microsoft::WRL::InhibitFtmBase;
+
+namespace text {
+    class dwrite_analysis_wrapper : public RuntimeClass<RuntimeClassFlags<ClassicCom | InhibitFtmBase>, IDWriteTextAnalysisSource, IDWriteTextAnalysisSink>
+    {
+    public:
+        dwrite_analysis_wrapper(const std::wstring& _text, const std::wstring& _userLocale) :
+            text(_text),
+            userLocale(_userLocale)
+        {
+        }
+
+#pragma region IDWriteTextAnalysisSource
+        HRESULT GetTextAtPosition(
+            UINT32 textPosition,
+            _Outptr_result_buffer_(*textLength) WCHAR const** textString,
+            _Out_ UINT32* textLength)
```

Comment:
```suggestion
            _Out_ UINT32* textLength) override
```

I think you are implementing a set of interface(s) here. Every function that is being implemented should be marked as `override`. (Or is there some Win32-specific thing that prevents that?)